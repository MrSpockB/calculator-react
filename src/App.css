.App {
  text-align: center;
}
.calculator {
  max-width: 300px;
  margin: 0 auto;
  margin-top: 40px;
}

.row {
  display: flex;
}

.op-btn, .num-btn {
  font-size: 25px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex: 1 0 25%;
  width: 25%;
  height: 70px;
  border: 1px solid;
  cursor: pointer;
}

.op-btn {
  color: white;
  background-color: rgb(221, 91, 91);
  border-color: black;
}

.num-btn {
  border-color: rgb(221, 91, 91);
}

.clear-button {
  display: flex;
  justify-content: center;
  align-items: center;
  flex: 2 0 50%;
  width: 50%;
  border: 1px solid rgb(221, 91, 91);
  font-size: 21px;
  color: #AAA;
  cursor: pointer;
}

.result {
  width: 100%;
  background-color: rgb(70, 56, 46);
  border: 1px solid rgb(221, 91, 91);
  color: white;
  height: 70px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding: 8px;
  font-size: 25px;
}
