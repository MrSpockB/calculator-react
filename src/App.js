import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>Calc</h1>
        <div className="calculator">
          <div className="row">
            <div className="result">0</div>
          </div>
          <div className="row">
            <div className="clear-button">clear</div>
            <div className="op-btn">/</div>
          </div>
          <div className="row">
            <div className="num-btn">7</div>
            <div className="num-btn">8</div>
            <div className="num-btn">9</div>
            <div className="op-btn">-</div>
          </div>
          <div className="row">
            <div className="num-btn">4</div>
            <div className="num-btn">5</div>
            <div className="num-btn">6</div>
            <div className="op-btn">+</div>
          </div>
          <div className="row">
            <div className="num-btn">1</div>
            <div className="num-btn">2</div>
            <div className="num-btn">3</div>
            <div className="op-btn">=</div>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
