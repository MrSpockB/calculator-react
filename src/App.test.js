import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders a snapshot', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should calculate correctly 2 + 2', () => {
  const component = renderer.create(<App/>);
  component.getInstance().onNumButtonClick('2');
  component.getInstance().setOperation('+');
  component.getInstance().onNumButtonClick('2');
  component.getInstance().onEqualBtnClick();
  expect(component.toJSON()).toMatchSnapshot();
});

it('should calculate correctly 4 - 2', () => {
  const component = renderer.create(<App/>);
  component.getInstance().onNumButtonClick('4');
  component.getInstance().setOperation('-');
  component.getInstance().onNumButtonClick('2');
  component.getInstance().onEqualBtnClick();
  expect(component.toJSON()).toMatchSnapshot();
});

it('should calculate correctly 12 * 2', () => {
  const component = renderer.create(<App/>);
  component.getInstance().onNumButtonClick('12');
  component.getInstance().setOperation('*');
  component.getInstance().onNumButtonClick('2');
  component.getInstance().onEqualBtnClick();
  expect(component.toJSON()).toMatchSnapshot();
});

it('should calculate correctly 100 / 25', () => {
  const component = renderer.create(<App/>);
  component.getInstance().onNumButtonClick('100');
  component.getInstance().setOperation('/');
  component.getInstance().onNumButtonClick('25');
  component.getInstance().onEqualBtnClick();
  expect(component.toJSON()).toMatchSnapshot();
});

it('should clear entered numbers and display 0', () => {
  const component = renderer.create(<App/>);
  component.getInstance().onNumButtonClick('500');
  component.getInstance().setOperation('+');
  component.getInstance().onNumButtonClick('3491');
  component.getInstance().onClearBtnClick();
  expect(component.toJSON()).toMatchSnapshot();
});
