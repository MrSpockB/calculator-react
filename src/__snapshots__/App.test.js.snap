// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <div
    className="calculator"
  >
    <div
      className="row"
    >
      <div
        className="result"
      >
        0
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="clear-button"
        onClick={[Function]}
      >
        clear
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        /
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        *
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        7
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        8
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        9
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        -
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        5
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        6
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        +
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        =
      </div>
    </div>
  </div>
</div>
`;

exports[`should calculate correctly 2 + 2 1`] = `
<div
  className="App"
>
  <div
    className="calculator"
  >
    <div
      className="row"
    >
      <div
        className="result"
      >
        4
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="clear-button"
        onClick={[Function]}
      >
        clear
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        /
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        *
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        7
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        8
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        9
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        -
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        5
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        6
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        +
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        =
      </div>
    </div>
  </div>
</div>
`;

exports[`should calculate correctly 4 - 2 1`] = `
<div
  className="App"
>
  <div
    className="calculator"
  >
    <div
      className="row"
    >
      <div
        className="result"
      >
        2
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="clear-button"
        onClick={[Function]}
      >
        clear
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        /
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        *
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        7
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        8
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        9
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        -
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        5
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        6
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        +
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        =
      </div>
    </div>
  </div>
</div>
`;

exports[`should calculate correctly 12 * 2 1`] = `
<div
  className="App"
>
  <div
    className="calculator"
  >
    <div
      className="row"
    >
      <div
        className="result"
      >
        24
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="clear-button"
        onClick={[Function]}
      >
        clear
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        /
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        *
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        7
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        8
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        9
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        -
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        5
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        6
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        +
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        =
      </div>
    </div>
  </div>
</div>
`;

exports[`should calculate correctly 100 / 25 1`] = `
<div
  className="App"
>
  <div
    className="calculator"
  >
    <div
      className="row"
    >
      <div
        className="result"
      >
        4
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="clear-button"
        onClick={[Function]}
      >
        clear
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        /
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        *
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        7
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        8
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        9
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        -
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        5
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        6
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        +
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        =
      </div>
    </div>
  </div>
</div>
`;

exports[`should clear entered numbers and display 0 1`] = `
<div
  className="App"
>
  <div
    className="calculator"
  >
    <div
      className="row"
    >
      <div
        className="result"
      >
        0
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="clear-button"
        onClick={[Function]}
      >
        clear
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        /
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        *
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        7
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        8
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        9
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        -
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        5
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        6
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        +
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="num-btn"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="num-btn"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="op-btn"
        onClick={[Function]}
      >
        =
      </div>
    </div>
  </div>
</div>
`;
